### DISCUSSION TITLE
Muti tenant install pattern of Argocd

### DISCUSSION THREAD

**Discussion by sreejesh123** [argoproj-labs/argocd-operator#1360] [Q&A]
Hi,  

we are running a multi tenant (hybrid) cloud with around 100 tenants , we are considering to use argocd for deployment, 

What  is the design pattern for installing Argocd on a multi tenant platform like ours? 

we are considering to install ONE ARGOCD per tenants instead of a ONE LARGE MONOLITIIC argocd per cluster for all tenants.

  Just wanted to know if you have any recommendations. And any drawbacks (other than mainrianace)  for running multiple argocd?

**Comment by davidt-gh** [comment]
Not just maintaining is what you need.
Every argocd need to have configuration, if it's app-of-apps it's onetime starting, then you getting all the app.
If you use multi-argocd, you will need to make sure it's all configured right in all of them.
Better use one ArgoCD for all tenants, one ArgoCD can manage multi cluster, and when specific application, you can specify which cluster is used for each app.
